src/vmm/src/vstate/vcpu/mod.rs

```diff
@@ -608,12 +622,21 @@ impl VcpuHandle {
     }
 }
 
+#[derive(Debug, Copy, Clone)]
+#[repr(u8)]
 pub enum VcpuEmulation {
     Handled,
     Interrupted,
     Stopped,
 }
 
+impl PartialEq for VcpuEmulation {
+    fn eq(&self, other: &Self) -> bool {
+        let (x, y) = (self, other);
+        *x as u8 == *y as u8
+    }
+}
```

Comment:
I think you can directly derive `PartialEq`:

```suggestion
#[derive(Debug, Copy, Clone, PartialEq)]
pub enum VcpuEmulation {
    Handled,
    Interrupted,
    Stopped,
}
```